### PR TITLE
Ruff

### PR DESCRIPTION
**Pull request by Cadair**
I don't quite know how I ended up here...

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -83,3 +83,81 @@ test-requires = ["opencv_python==4.6.0.66"]
       directory = "trivial"
       name = "Internal Changes"
       showcontent = true
+
+
+[tool.ruff]
+target-version = "py38"
```
Comment (nabobalis) [inline pyproject.toml:89]:
```suggestion
target-version = "py39"
```

**Comment by ayshih** [comment]
I also don't like the changes to `_supported_observer_coordinates()` in the map sources

**Comment by nabobalis** [comment]
> I also don't like the changes to `_supported_observer_coordinates` in the map sources

I assume that is a side effect of ignoring the line length which looks like it has  TODO attached to?

**Comment by ayshih** [comment]
Sure.  It's amusing, albeit understandable, that telling an automatic code formatter to ignore line length gives it license to make heinously long lines.

**Comment by Cadair** [comment]
I don't think ruff autofixes line too long anyway.

**Comment by nabobalis** [comment]
With the three ruff PRS merged, rebasing this might be a pain and a fresh PR with extra config should be opened.

**Pull request closed without merging**